javascript/javascript1/week4/voice-assistant/voice-assistant.js

```diff
@@ -0,0 +1,93 @@
+const dayjs = require('dayjs');
+const math = require('mathjs');
+
+function getReply(command) {
+  const storeName = "Benjamin";
+  const toDo = ["fishing", "singing in the shower"];
+  const myDate = dayjs();
+  const formatOfDate = myDate.format("dddd DD of MMMM YYYY, hh:mm A");
+  
+  switch (command) {
+    case `Hello my name is ${storeName}`:
+      return `Nice to meet you ${storeName}`;
+
+    case "What is my name?":
+      if (storeName.toLowerCase) {
+        return `Your name is ${storeName}`;
+      } else {
+        return "Your name has not been set yet";
+      }
+
+    case "What day is it today?":
+      return `Today is ${formatOfDate}`;
+
+    default:
+      if (command.startsWith("Add ")) {
+        const task = command.replace("Add ", "").replace(" to my todo", "");
+        toDo.push(task);
+        return `${task} added to your todo`;
+      }
+
+      if (command.startsWith("Remove ")) {
+        const task = command
+          .replace("Remove ", "")
+          .replace(" from my todo", "");
+        const index = toDo.indexOf(task);
+        if (index > -1) {
+          toDo.splice(index, 1);
+          return `Removed ${task} from your todo`;
+        } else {
+          return `${task} not found in your todo`;
+        }
+      }
+
+      if (command === "What is on my todo?") {
+        if (toDo.length === 0) {
+          return "Your todo list is empty";
+        } else {
+            return `You have ${toDo.length} todos on your list`
+        }
+      }
+      
+      if (command.startsWith("Calculate ")) {
+        const calculate = command.replace("Calculate ", "");
+        try {
+          const result = math.evaluate(calculate);
+          return `The result of ${calculate} is ${result}`;
+        } catch (error) {
+          return "Sorry, I didn't catch that";
+        }
```

Comment:
Really great to see you're not afraid to read up on and try out other libraries!

No point re-inventing the wheel. Take the shortcuts you can!